document-generator-core/swagger-2.0/models/document-generator.json

```diff
@@ -0,0 +1,76 @@
+{
+  "definitions": {
+    "documentGeneratorRequest": {
+      "title": "DocumentGeneratorRequest",
+      "required": [
+        "id",
+        "resource",
+        "resource_id",
+        "content_type",
+        "document_type",
+        "user_id"
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "id for the document request"
+        },
+        "resource": {
+          "type": "string",
+          "description": "link to the resource"
+        },
+        "resource_id": {
+          "type": "string",
+          "description": "id of the resource"
+        },
+        "content_type": {
+          "type": "string",
+          "description": "the content type of the template to be converted"
+        },
+        "document_type": {
```

Comment:
`document_type` in the body is confusing, this denotes the format that the document should be generated in so we should use `mime_type` and enumerate the options we support which will currently be `application/pdf` and `text/html` (for html5 and ixbrl documents)